cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -231,4 +245,63 @@ public String getValue() {
       }
     };
   }
+
+  public static Map<String, String> getMetricsTags(LoggingContext context) throws IllegalArgumentException {
+    if (context instanceof ServiceLoggingContext) {
+     return getTagsFromSystemContext((ServiceLoggingContext) context);
+    } else {
+      return getMetricsTagsFromLoggingContext(context);
+    }
+  }
+
+  private static Map<String, String> getMetricsTagsFromLoggingContext(LoggingContext context) {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    Map<String, LoggingContext.SystemTag> loggingTags = context.getSystemTagsMap();
+    String namespace = getValueFromTag(loggingTags.get(NamespaceLoggingContext.TAG_NAMESPACE_ID));
+
+    if (namespace == null || namespace.isEmpty()) {
+      throw new IllegalArgumentException("Cannot find namespace in logging context");
+    }
+    builder.put(Constants.Metrics.Tag.NAMESPACE, namespace);
+
+    String applicationId = getValueFromTag(loggingTags.get(ApplicationLoggingContext.TAG_APPLICATION_ID));
+    String adapterId = getValueFromTag(loggingTags.get(ApplicationLoggingContext.TAG_ADAPTER_ID));
+    // Must be an application or adapter
+    if (applicationId == null && adapterId == null) {
+      throw new IllegalArgumentException("Missing application or adapter id");
+    }
+    if (applicationId != null) {
+      builder.put(Constants.Metrics.Tag.APP, applicationId);
+      LoggingContext.SystemTag entityId = getEntityId(context);
+      String entityName = getMetricsTagNameFromLoggingContext(entityId);
+      if (entityName != null) {
+        builder.put(entityName, entityId.getValue());
+      }
+    } else {
+      builder.put(Constants.Metrics.Tag.ADAPTER, adapterId);
+    }
+    return builder.build();
+  }
+
+
+  private static String getMetricsTagNameFromLoggingContext(LoggingContext.SystemTag tag) {
+    return LOG_TAG_TO_METRICS_TAG_MAP.get(tag.getName());
+  }
+
+  private static String getValueFromTag(LoggingContext.SystemTag tag) {
+    if (tag == null) {
+      return null;
+    } else {
+      return tag.getValue();
+    }
+  }
+
+
+  public static Map<String, String> getTagsFromSystemContext(ServiceLoggingContext context) {
```

Comment:
To be uniform with `getMetricsTagsFromLoggingContext`, call this method `getMetricsTagsFromSystemContext`?